### DISCUSSION TITLE
How to get a list of all l10n keys used by a calendar?

### DISCUSSION THREAD

**Discussion by tukusejssirs** [romcal/romcal#454] [Q&A]
@emagnier, I wanted to create a unit test to check if all l10n keys are used in at least one calendar and if all used l10n keys in any calendar are defined in `en`.

I struggle with the first step: how to get a list of all l10n keys used by a calendar?

I thought I could initialise the calendar function (e.g. `new Americas(config)`, however, I have been not able to make it work (what should be the config? what properties are required?) and it might contain only celebrations celebrated in a particular year.

Basically, all I need is the `$calendar.inputs` value.

Any help appreciated! :pray:

**Comment by emagnier** [comment]
You must first get all the calendar keys: https://github.com/romcal/romcal#get-all-liturgical-day-definitions-getalldefinitions

Then you can get all related keys from the `i18nDef` property:

```js
const romcal = new Romcal(); // You can also specify a particular calendar to get more related keys.
const i18nKeys = Object.values(await romcal.getAllDefinitions()).map((def) => def.i18nDef);
```

(The last renaming of `keys` properties to `ids` for all types of entities is a bit confusing for me, especially when it comes to localization; here we usually refer to it as a `key`...)

You will also have to take into account all defined items from the martyrology.
